ovs/extensions/healthcheck/openvstorage/openvstoragecluster_health_check.py

```diff
@@ -89,45 +88,48 @@ def check_size_of_log_files(logger):
                     'checkLogfilesSize')
 
         # collect log files
         for log, settings in Helper.check_logs.iteritems():
-            if settings.get('type') == 'dir':
-                # check if dirname exists
-                if os.path.isdir(log):
-                    # check if dirname contains files
-                    files = OpenvStorageHealthCheck._list_logs_in_directory(log)
-                    # check if given dirname has files
-                    if len(files) != 0:
-                        # check size of log files
-                        for filename in files:
-                            if settings.get('prefix'):
-                                for prefix in list(settings.get('prefix')):
-                                    if prefix in filename:
-                                        collection.append(filename)
-                            else:
-                                collection.append(filename)
-
-                    # check if has nested_dirs and nested_files
-                    if settings.get('contains_nested'):
-                        nested_dirs = OpenvStorageHealthCheck._list_dirs_in_directory(log)
-                        for dirname in nested_dirs:
-                            nested_files = OpenvStorageHealthCheck._list_logs_in_directory(log+"/"+dirname)
-                            if len(nested_files) != 0:
-                                # check size of log files
-                                for nested_file in nested_files:
-                                    if settings.get('prefix'):
-                                        for prefix in list(settings.get('prefix')):
-                                            if prefix in filename:
-                                                collection.append(nested_file)
-                                    else:
-                                        collection.append(nested_file)
-            else:
+            if settings.get('type') != 'dir':
                 # check if filename exists
                 if os.path.exists(log):
                     collection.append(log)
+                continue
+            # check if dirname exists
+            if os.path.isdir(log) is False:
+                continue
+            # check if dirname contains files
+            files = OpenvStorageHealthCheck._list_logs_in_directory(log)
+            # check if given dirname has files
+            if len(files) != 0:
+                # check size of log files
+                for filename in files:
+                    if settings.get('prefix'):
+                        for prefix in list(settings.get('prefix')):
+                            if prefix in filename:
+                                collection.append(filename)
+                    else:
+                        collection.append(filename)
+
+            # check if has nested_dirs and nested_files
+            if settings.get('contains_nested') is False:
```

Comment:
In new scenario when ```settings``` does not contain the key ```contains_nested``` you will still execute code below, while in old scenario this wasn't the case